include/cppkafka/CMakeLists.txt

```diff
@@ -1,3 +1,17 @@
+# Local function to auto-generate main cppkafka.h header file
+function(make_cppkafka_header)
+    set(CPPKAFKA_HEADER ${CMAKE_CURRENT_SOURCE_DIR}/cppkafka.h)
+    execute_process(COMMAND rm -f ${CPPKAFKA_HEADER})
```

Comment:
I don't think this is portable. I think a better way to do this would be to have a default template file that has possibly the copyright and then a variable that you can set from here that will contain the block of includes.

We've done something similar in [another project](https://github.com/mfontanini/libtins/blob/master/include/tins/config.h.in), the file is generated [here](https://github.com/mfontanini/libtins/blob/master/CMakeLists.txt#L229).